File: src/p0997/python/solution.py
from typing import List


class Solution:
    def findJudge(self, n: int, trusts: List[List[int]]) -> int:
        degree = [0] * (n + 1)
        for trust in trusts:
            degree[trust[0]] -= 1
            degree[trust[1]] += 1
        for i in range(1, n + 1):
            if degree[i] == n - 1:
                return i
        return -1

File: src/p0997/java/solution.java
class Solution {
    public int findJudge(int N, int[][] trusts) {
        int[] degree = new int[N + 1];
        for (int[] trust : trusts) {
            degree[trust[0]]--;
            degree[trust[1]]++;
        }
        for (int i = 1; i <= N; i++) {
            if (degree[i] == N - 1) {
                return i;
            }
        }
        return -1;
    }
}
